src/strategies/MangroveOffer.sol

```diff
@@ -280,26 +283,34 @@ abstract contract MangroveOffer is AccessControlled, IOfferLogic {
       return "posthook/filled";
     }
     uint new_wants = __residualWants__(order);
-    try MGV.updateOffer(
-      order.outbound_tkn,
-      order.inbound_tkn,
-      new_wants,
-      new_gives,
-      order.offerDetail.gasreq(),
-      order.offerDetail.gasprice(),
-      order.offer.next(), // using next as pivot since this offer is off the book
+    return _updateOffer(
```

Comment:
Would it make sense to pass in order.offerDetail and order.global, order.local? This would get rid of the
```
(vars.global, vars.local) = MGV.config(address(args.outbound_tkn), address(args.inbound_tkn));
 vars.offerDetail = MGV.offerDetails(address(args.outbound_tkn), address(args.inbound_tkn), offerId);
```

in the _updateOffer function. This would save some gas for the happy path of reposting residual. For a call to updateOffer->_updateOffer from the outside we can then look these up in updateOffer.

The only drawback I see is that Direct has the same method signature and it does not need all that data.

Reply:
Let's keep it aside for future gas optim work (I think there are many spots where gas can be saved)